### PR TITLE
Enhancement/3244 ga4 submit changes

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue #3244 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/analytics-4/datastore/settings.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	if ( propertyID === PROPERTY_CREATE ) {
+		const accountID = select( MODULES_ANALYTICS ).getAccountID();
+		const { response: property, error } = await dispatch( STORE_NAME ).createProperty( accountID );
+		if ( error ) {
+			return { error };
+		}
+
+		propertyID = property._id;
+		await dispatch( STORE_NAME ).setPropertyID( propertyID );
+		await dispatch( STORE_NAME ).setWebDataStreamID( '' );
```
Comment (tofumatt) [inline assets/js/modules/analytics-4/datastore/settings.js:51]: Why do we need to set the webstream data ID as an empty string here? We select it and then check to see if it's valid, below, so presumably a `null`/`undefined` value or similar won't be valid, right?

Or does it need to be a string, but an invalid string, in order for the `createWebDataStream` path below to work? If so, it's worth adding a comment here to clarify that. At any rate it isn't obvious to me why this line is here.

**Review by tofumatt — CHANGES_REQUESTED** [review]
Looks good to me, but I have one question about the `setWebDataStreamID()` call.

After that's addressed this should be good to merge. 👍🏻

File: assets/js/modules/analytics-4/datastore/settings.js (hunk 1 above)
Comment (eugene-manuilov) [reply assets/js/modules/analytics-4/datastore/settings.js:52]: We need to set an invalid web data stream ID to make sure a new data stream will be created. I have added a comment to explain it.

**Review by tofumatt — APPROVED** [review]
Thanks for the explanation on the empty string for the web stream ID! Looks good 👍🏻

**Pull request merged**